code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -81,11 +81,13 @@
 			else
 				to_chat(H, span("danger", "You sense great disturbance to your physical body!"))
 		else
-			user.visible_message("<span class='danger'>[user] attacked [src] with [I] waking [t_him] up!</span>", \
-								"<span class='danger'>You attacked [src] with [I], but they do not respond... Maybe they have S.S.D?</span>")
+			visible_message("<span class='danger'>[user] attacked [src] with [I] waking [t_him] up!</span>")
+			if(user.client)
+				to_chat(user, "<span class='danger'>You attacked [src] with [I], but they do not respond... Maybe they have S.S.D?</span>")
 	else if(client && willfully_sleeping)
-		user.visible_message("<span class='danger'>[user] attacked [src] with [I] waking [t_him] up!</span>", \
-							"<span class='danger'>You attacked [src] with [I] waking [t_him] up!</span>")
+		visible_message("<span class='danger'>[user] attacked [src] with [I] waking [t_him] up!</span>",)
```

Comment:
Why is this change necessary? The only thing potentially wrong about the old code is that `user` might be null, but if that is not the case, then the old code should work fine and is 100x more readable.

Reply:
That is the case, `user` becomes null. So it crashes and does get to lines which wake them up.

Reply:
Then this fix will still crash. Because if `user == null` then doing `if (user.client)` will run to a null exception.

Keep the original line, since it is more concise, just add a check for user into the `else if`. So line 87 would become `else if (client && user && willfully_sleeping)`

Reply:
Sorry. I meant when `user.client` is null. Hence attacking mobs with no client which are any simple mobs

Reply:
This runtime doesn't make sense, then. `mob/visible_message` is safe to call even when the client of the mob is absent, thus, the original syntax is completely fine.

Can you post the actual stack of the runtime?